meta.go

```diff
@@ -650,6 +695,26 @@ func (meta Meta) Check() error {
 		names[name] = true
 	}
 
+	for name, device := range meta.Device {
+		if device.Type == "" {
+			return fmt.Errorf("charm %q device %q: type must be specified", meta.Name, name)
+		}
+		if err := device.Type.Validate(); err != nil {
+			return err
+		}
+		if device.CountMin <= 0 {
+			return fmt.Errorf("charm %q device %q: invalid countmin amount %d", meta.Name, name, device.CountMin)
```

Comment:
Let's do the error messages the same as for storage:

charm %q storage %q: invalid minimum count %d

Also 0 is valid as a min so the check is < 0

And as per what's done for storage we also need to check for duplicate names